akka-http/src/main/scala/akka/http/javadsl/server/Directives.scala

```diff
@@ -26,4 +28,96 @@ object Directives extends AllDirectives {
 
   @varargs override def getFromBrowseableDirectories(directories: String*): Route =
     super.getFromBrowseableDirectories(directories: _*)
+
+  /**
+   * Partially applying functions for the Java World!
```

Comment:
Maybe in [`akka-http/akka.http.javadsl.common`](https://github.com/akka/akka-http/tree/master/akka-http/src/main/java/akka/http/javadsl/common)`.JavaFunctions`? @ktoso what do you think?